### DISCUSSION TITLE
Optimize me #3,  MotionSensor

### DISCUSSION THREAD

**Discussion by SomeRandomNoobKekeke** [FakeFishGames/Barotrauma#15206] [Ideas and suggestions]
Ackshually your implementation of aabb checks is fine, the laggy part is:
https://github.com/FakeFishGames/Barotrauma/blob/7d5b7a310a52372dac0a8bac90fd2a6a2ab342dc/Barotrauma/BarotraumaShared/SharedSource/Items/Components/Signal/MotionSensor.cs#L192
That's unexpected (for me), but i'm not yapping about it because i don't see how to fix it, might require another signal system rework
![20241120080057_1](https://github.com/user-attachments/assets/39b17394-825e-422f-a801-fea0356443a7)
![20241120080105_1](https://github.com/user-attachments/assets/cff8a43a-ff17-447b-b759-69c073c424d2)

**Comment by Regalis11** [comment]
Thank you for the report! I'm not sure there's much that can be done about this, aside from reworking the entire signal system in some way. The SendSignal method itself here isn't the main culprit, the costly part is making the signal propagate through the circuits and making the recipient items react to it.

We could potentially attempt something like making the sensors only send out the value periodically or when the value changes, but that would cause backwards compatibility issues in older subs. Maybe introducing that as a new option could be worth a shot though.
